graphql/graphql.go

```diff
@@ -902,11 +935,14 @@ type Resolver struct {
 }
 
 func (r *Resolver) Block(ctx context.Context, args struct {
-	Number *hexutil.Uint64
+	Number *Long
 	Hash   *common.Hash
 }) (*Block, error) {
 	var block *Block
 	if args.Number != nil {
+		if *args.Number < 0 {
+			return nil, nil
+		}
```

Comment:
That's not in the EIP: https://eips.ethereum.org/EIPS/eip-1767
You get the same behavior without using -1, since it's the default to offer the latest block when no block number is passed.
https://github.com/hyperledger/besu/blob/9e224ed7f3a90677aa59553a770223cd6f2dd3d0/ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java#L165 doesn't indicate that -1 gets special treatment. It goes straight to DB after that.

Reply:
Ok